docs/contribution.md

```diff
@@ -13,9 +13,25 @@ The pull requests to read only subtree split [repositories](https://github.com/p
 
 ## Setup environment
 
+PR's go through automatic testing and code style checks. For your convenience 3 commands below will set up your local
+environment. You will need:
+1. A globally accessible `composer` script for package dependencies
+2. Docker & `docker-compose` for running tests against services
+
+Note it is assumed your `docker-compose` is available without sudo.
+
+First install dependencies:
+```
+composer install --ignore-platform-reqs
```

Comment:
Could you please explain why would we need `--ignore-platform-reqs`? Looks dangerous for me

Reply:
It's only a suggestion for local development in cases where not every extension is installed (for example, you might be missing `ext-mongo` or `ext-rdkafka`). Otherwise you're forced to install theoretically unrelated extensions, even if you only intend to work on a specific subset of enqueue-dev functionality.

Reply:
this is what happens when the project is not split by different packages with carefully thought dependencies :(

Reply:
It's a concept of monorepo that is not really uncommon. We can discuss it's merits and demerits, but you can encounter it in a couple of different open source projects, like [Symfony](https://github.com/symfony/symfony/blob/master/composer.json).

In this case, it allows all packages to inherit the main structure and be tested inside one, main project. This makes cross-transport changes **a lot** easier.

Eventually, subpackages are split to subrepositories for consumption in packagist.

Reply:
In fact, you don't need `--ignore-platform-reqs`. enqueue-dev use compose platforms feature to install deps and runs tests in prepared containers where all extensions and services are in place. 

Split repos have correct deps too

Reply:
> In fact, you don't need `--ignore-platform-reqs`. enqueue-dev use compose platforms feature to install deps and runs tests in prepared containers where all extensions and services are in place.

@makasim true, but only to an extent - contribution docs specifically say what has to be done to get enqueue-dev to install dependencies for local, non-containerized environment. That's why I suggested to add `--ignore-platform-reqs` for it, in cases where someone wants to work on a specific transport and doesn't really want to install other extensions that he/she doesn't care about.

I could convert it into a note instead or drop it entirely, working on assumption that developer will have enough knowledge to handle it himself.

Reply:
This https://github.com/php-enqueue/enqueue-dev/blob/master/composer.json#L110-L120 should do for local setup.

Reply:
Ouch. For some reason my local setup had a lock file that seemed to not include those platform options :disappointed: I'll remove it.